static/js/compose_actions.js

```diff
@@ -309,7 +309,7 @@ exports.respond_to_message = function (opts) {
     } else {
         msg_type = message.type;
     }
-    exports.start(msg_type, {stream: stream, subject: subject,
```

Comment:
How does the start function work?
I mean the {stream: stream, subject: subject...
Is it creating an object?
Just a curiosity question :)

Reply:
I did not really look into it when doing the refactoring since I only changed the variable name. But yes I think creating an object is what it does. Will have to look into it more when updating the subject field name to topic.

Reply:
What it does is it creates an object, which basically means that it passes a JSON object as a parameter to the function. The JSON is therefore an object with **keys** stream, subject, private_message_recipient, replying_to_message and **values** stream, topic, pm_recipient, message, opts.trigger respectively.

Reply:
Similarly you could deconstruct an object passed to a function like [this](https://simonsmith.io/destructuring-objects-as-function-parameters-in-es6/). Although that is an ES6 ("new" JS version) function and they do not use ES6 in this project.

Reply:
@jockeboman @eywoon

Reply:
That's cool!
Especially the ES6 way.
Thanks for the explanation.

Reply:
Okay thanks for explaining!